packages/location-management/src/components/LocationForm/index.tsx

```diff
@@ -372,7 +372,9 @@ const LocationForm = (props: LocationFormProps) => {
               showSearch
               placeholder={ENTER_A_LOCATION_GROUP_NAME_PLACEHOLDER}
               loadData={(setData) => {
-                return loadLocationTags(openSRPBaseURL, service, setData);
+                return loadLocationTags(openSRPBaseURL, service).then((response) =>
+                  setData(response)
+                );
```

Comment:
@ashfahan what necessiated this change?

Reply:
it was a refractor i pull out from other pr while i was working on keyclock user this way it was easier to predict the code flow.

Reply:
I think its unnecessary, does not seem to significantly enhance stuff, and has the possibility of causing a regression in other places where its being used.

Reply:
if you have to do this, I would say make the callback optional, and then modify the loader function such that it returns the resonse if the callbac is not defined. THis way its still backwards compatible,